shared/tools/base.py

```
@@ -0,0 +1,385 @@
+        """データベース接続のコンテキストマネージャー"""
+        if not self.require_db:
+            yield None
+            return
+            
+        connection = None
+        try:
+            connection = await connection_pool.get_connection()
+            yield connection
+        except Exception as e:
+            logger.error(f"DB接続エラー: {e}")
+            raise DatabaseError(f"データベース接続に失敗しました: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)